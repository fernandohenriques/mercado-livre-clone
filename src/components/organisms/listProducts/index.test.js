import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import ListProducts from './index';

describe('<ListProducts />', () => {
  const productsMock = [
    {
      id: '1',
      title: 'Kit de Perfumes',
      price: {
        currency: 'R$',
        amount: 50,
      },
      picture: 'https://images.mercadolivre.com.br/imgone.png',
      free_shipping: false,
      address: {
        state_name: 'São Paulo',
      },
    },
    {
      id: '2',
      title: 'Tênis All Star',
      price: {
        currency: 'R$',
        amount: 60,
      },
      picture: 'https://images.mercadolivre.com.br/imgtwo.png',
      free_shipping: true,
      address: {
        state_name: 'Rio de Janeiro',
      },
    },
  ];

  it('Snapshot testing', () => {
    const { asFragment } = render(<ListProducts products={productsMock} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render ProductCard component', () => {
    const wrapper = shallow(<ListProducts products={productsMock} />);

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.find('ProductCard')).toHaveLength(2);
  });
});
