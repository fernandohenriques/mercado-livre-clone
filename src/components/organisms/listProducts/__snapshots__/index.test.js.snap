// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListProducts /> Snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="card"
    data-id="1"
  >
    <img
      alt="Kit de Perfumes"
      src="https://images.mercadolivre.com.br/imgone.png"
      title="Kit de Perfumes"
    />
    <div
      class="data"
    >
      <div
        class="priceAndTitle"
      >
        <div
          class="price"
        >
          $ 50.00
        </div>
        <span
          class="title"
        >
          Kit de Perfumes
        </span>
      </div>
      <div
        class="state"
      >
        São Paulo
      </div>
    </div>
  </div>
  <div
    class="divider"
  />
  <div
    class="card"
    data-id="2"
  >
    <img
      alt="Tênis All Star"
      src="https://images.mercadolivre.com.br/imgtwo.png"
      title="Tênis All Star"
    />
    <div
      class="data"
    >
      <div
        class="priceAndTitle"
      >
        <div
          class="price"
        >
          $ 60.00
          <img
            src="ic-shipping.png"
          />
        </div>
        <span
          class="title"
        >
          Tênis All Star
        </span>
      </div>
      <div
        class="state"
      >
        Rio de Janeiro
      </div>
    </div>
  </div>
</DocumentFragment>
`;
